### ISSUE TITLE
Boolean feature columns are dropped from the design matrix instead of cast to 0/1 (config declares them, model never sees them)

### ISSUE THREAD

**Issue by FerSobrino** [ccd-ia/triage-pg#12]
_Context: NHL game-outcome prediction on triage-pg 1.1.5 + featurizer 1.1.0 (HockeyPredicts rebuild). Filed from our engineering log._

## 9. Non-numeric feature columns are dropped from the design matrix, so a PostgreSQL `boolean` feature silently disappears

**Repo:** `ccd-ia/triage-pg` · **Version:** 1.1.5 · **Severity:** medium (data loss, warned but easy to miss)

**What happens.** `matrix.py:_numeric_feature_columns` filters the design frame to
numeric dtypes before the Parquet is written, and logs

```
dropping 6 non-numeric feature column(s) — [('knowledge_date', 'Date'),
('away_situational__b2b', 'Boolean'), ('away_situational__third_in_four', 'Boolean'),
('home_situational__b2b', 'Boolean'), ('home_situational__third_in_four', 'Boolean'),
('situational__matinee', 'Boolean')]. sklearn needs a numeric design matrix; …
```

The reasoning is sound — sklearn needs a numeric matrix — and the warning is
real. But a **boolean** is not like the other non-numeric types: it is already a
0/1 feature, `numpy`/`polars` cast it for free, and every estimator in the grid
handles it. Declaring it `type: numeric` in the featurizer config does not help:
the declared type does not change what PostgreSQL returns, so the column arrives
as `Boolean` and is dropped anyway.

The failure mode is quiet in the way that matters: `triage analyze-config` counts
the column (it is in featurizer's plan), the config declares it, the run trains —
and the model simply never sees it. Our five situational flags would have gone
missing from a matrix the config claimed had 136 features.

**Expected.** Cast `Boolean` to a 0/1 numeric column instead of dropping it (and
keep dropping `Date` / `String`, which genuinely need a user decision). Failing
that, make a dropped **boolean** an error rather than a warning, since it is
always recoverable and never intentional.

Second, smaller point visible in the same message: `knowledge_date` — the target
entity's own `temporal_ix`, never declared as a variable — is emitted by
featurizer as a feature column and then dropped here. The warning's advice
("mark it role:identifier in the featurizer config") cannot be followed, because
the column is not in the `variables:` map to mark.

**Our workaround.** Declare each boolean as

```yaml
home_situational__b2b:
  type: boolean
  role: categorical
  vocabulary: ['true', 'false']
```

so featurizer one-hot encodes it into `game_wide.home_situational__b2b=true` /
`=false` — numeric, and therefore kept. Cost: 5 declared variables become 10
matrix columns (136 declared → 141 planned), the one-hot names carry a
`game_wide.` prefix the plain passthroughs do not, and every positional
hyperparameter (`EloLogisticClassifier.feature_index`) and name list
(`AttentionMLPClassifier.feature_names`) has to be computed over the 141, not the
136. All three are pinned by `tests/triage/test_real_experiment_configs.py`.
